frida_mode/src/cmplog/cmplog.c

```diff
@@ -35,63 +49,76 @@ static gint cmplog_sort(gconstpointer a, gconstpointer b) {
 
 }
 
-static int cmplog_create_temp(void) {
+static void cmplog_get_ranges(void) {
 
-  const char *tmpdir = g_get_tmp_dir();
-  OKF("CMPLOG Temporary directory: %s", tmpdir);
-  gchar *fname = g_strdup_printf("%s/frida-cmplog-XXXXXX", tmpdir);
-  OKF("CMPLOG Temporary file template: %s", fname);
-  int fd = mkstemp(fname);
-  OKF("CMPLOG Temporary file: %s", fname);
+  guint64 current_time = g_get_monotonic_time();
+  if (range_data->time != 0) {
 
-  if (fd < 0) {
-
-    FATAL("Failed to create temp file: %s, errno: %d", fname, errno);
+    if (current_time - range_data->time < RANGE_DATA_FREQ) { return; }
 
   }
 
-  if (unlink(fname) < 0) {
+  OKF("CMPLOG - Collecting ranges");
+
+  GArray *cmplog_ranges =
+      g_array_sized_new(false, false, sizeof(GumMemoryRange), 100);
+  gum_process_enumerate_ranges(GUM_PAGE_READ, cmplog_range, cmplog_ranges);
+  g_array_sort(cmplog_ranges, cmplog_sort);
+
+  if (cmplog_ranges->len > RANGE_DATA_MAX) {
 
-    FATAL("Failed to unlink temp file: %s (%d), errno: %d", fname, fd, errno);
+    FATAL("Too many ranges: %u > %lu", cmplog_ranges->len, RANGE_DATA_MAX);
```

Comment:
why do you FATAL? is this necessary? you make a target unfuzzable with that. if possible just work with the number of ranges you can handle and live with that that you will miss stuff.